pymc_marketing/mlflow.py

```diff
@@ -528,12 +928,29 @@ def autolog(
     """
     arviz_summary_kwargs = arviz_summary_kwargs or {}
 
+    def patch_compute_log_likelihood(compute_log_likelihood):
+        """Patch the compute_log_likelihood function to log LOOCV metrics.
+
+        We don't need to call this directly again in MMM.fit patch, since that function calls
+        pm.sample() internally anyway.
+        """
+
+        @wraps(compute_log_likelihood)
+        def new_compute_log_likelihood(*args, **kwargs):
+            idata = compute_log_likelihood(*args, **kwargs)
+            log_loocv_metrics(idata=idata)
+
+            return idata
+
+        return new_compute_log_likelihood
+
+    pm.compute_log_likelihood = patch_compute_log_likelihood(pm.compute_log_likelihood)
+
     def patch_sample(sample):
         @wraps(sample)
         def new_sample(*args, **kwargs):
+            model = pm.modelcontext(kwargs.get("model"))
             idata = sample(*args, **kwargs)
-            mlflow.log_param("pymc_marketing_version", __version__)
-            mlflow.log_param("pymc_version", pm.__version__)
```

Comment:
Can we keep for *all* pymc models

Reply:
Ah yep my mistake - I was getting confused with how the scope of the pm.sample works after being patched and then used in mmm.fit().

Should be fixed now I hope!